src/main/java/org/team1540/robot2018/subsystems/DriveTrain.java

```diff
@@ -0,0 +1,105 @@
+package org.team1540.robot2018.subsystems;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import org.team1540.base.ChickenSubsystem;
+import org.team1540.base.wrappers.ChickenTalon;
+import org.team1540.robot2018.RobotMap;
+import org.team1540.robot2018.Tuning;
+import org.team1540.robot2018.commands.JoystickDrive;
+
+public class DriveTrain extends ChickenSubsystem {
+
+  private ChickenTalon left = new ChickenTalon(RobotMap.DRIVE_LEFT);
+  private ChickenTalon left2 = new ChickenTalon(RobotMap.DRIVE_LEFT_2);
+  private ChickenTalon left3 = new ChickenTalon(RobotMap.DRIVE_LEFT_3);
+  private ChickenTalon[] lefts = new ChickenTalon[]{left, left2, left3};
+  private ChickenTalon right = new ChickenTalon(RobotMap.DRIVE_RIGHT);
+  private ChickenTalon right2 = new ChickenTalon(RobotMap.DRIVE_RIGHT_2);
+  private ChickenTalon right3 = new ChickenTalon(RobotMap.DRIVE_RIGHT_3);
+  private ChickenTalon[] rights = new ChickenTalon[]{right, right2, right3};
+  private ChickenTalon[] talons = new ChickenTalon[]{left, left2, left3, right, right2, right3};
+  private ChickenTalon[] masters = new ChickenTalon[]{left, right};
+
+  public DriveTrain() {
+    this.add(talons);
+    this.setPriority(10);
+
+    left.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+    right.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+
+    left.setSensorPhase(true);
+
+    for (ChickenTalon talon : lefts) {
+      talon.setInverted(false);
+    }
+
+    right.setSensorPhase(true);
+
+    for (ChickenTalon talon : rights) {
+      talon.setInverted(true);
+    }
+
+    left2.set(ControlMode.Follower, left.getDeviceID());
+    left3.set(ControlMode.Follower, left.getDeviceID());
+
+    right2.set(ControlMode.Follower, right.getDeviceID());
+    right3.set(ControlMode.Follower, right.getDeviceID());
+
+    for (ChickenTalon talon : talons) {
+      talon.setBrake(true);
+    }
+  }
+
+  @Override
+  public void initDefaultCommand() {
+    setDefaultCommand(new JoystickDrive());
+  }
+
+  public void setLeft(double value) {
+    this.left.set(ControlMode.PercentOutput, value);
+  }
+
+  public void setRight(double value) {
+    this.right.set(ControlMode.PercentOutput, value);
+  }
+
+  @Override
+  public void periodic() {
+    for (ChickenTalon talon : masters) {
```

Comment:
Are these really changing so often they need to be constantly set?

Reply:
Little to no overhead, and allows them to be tuned on the fly.

Reply:
Why not just have an update method that gets called when your tunables are modified?

Reply:
You'd have to do constant edge detection since you can't really have proper listeners due to the way NetworkTables (and by extension SmartDashboard and by extension the `AdjustableManager`) works. At that point, it's a good bit more code for a negligible (maybe nonexistent or negative if CTRE's software checks if values are the same before sending them) performance benefit.

Reply:
Maybe I'm missing something on how NetworkTables work, but couldn't you just trip a flag in the NetworkTable and then run all the updating? Then you're only constantly checking one value.

Reply:
Also you could spin that off into another thread and have it sleep a bunch.

Reply:
Although then locking ew.

Reply:
Nope. The `AdjustableManager` is literally just an annotation-based wrapper around the `SmartDashboard.putX` and `SmartDashboard.getX` functions. Having a boolean flag set when a value is modified would basically entail writing our own SmartDashboard that supported that sort of thing, because SmartDashboard right now is just a dressed-up Outline Viewer with graphs.

Reply:
If you decide tho to do grouping for all the organization stuff (earlier comment) anyways, then you've got a convenient group with which to use your flag.

Reply:
Fair, but that would also require changes on the library level to be effective and I'm not sure if SmartDashboard supports "modified" flags anyway.